### PR TITLE
Visual Tweaks

### PR DESCRIPTION
**Pull request by Ericzklm**
Added move type icons and item icons to dropdown

### CODE REVIEW

**Comment by Ericzklm** [comment]
Added icons to some of the dropdown menus; is this a change that you'd be interested in? 
image
image
The moves dropdown gets a little cluttered

**Comment by Ericzklm** [comment]
Now that I think about it, each icon is its own request to the Github content server which could be an issue if there are limits. Not sure how that works to be honest.

**Comment by theastrogoth** [comment]
I think the item sprites look great! I agree about the move dropdown getting a bit cluttered, and type icons seem a little redundant for moves that have a TM disc.

Another idea for additional icons: displaying Pokemon sprites (currently not used here) in the dropdowns used in the Raid Move Selection UI, similar to the way your graphics have used them.  We're a little pressed for space there at the moment, so it might take widening that area or moving some things around.

> Now that I think about it, each icon is its own request to the Github content server which could be an issue if there are limits. Not sure how that works to be honest.

I've checked my "Plans and usage" on Github and everything is still sitting at exactly zero, so it doesn't seem like we're going to have any issues there. It looks like there is a soft limit on bandwidth of 100GB/month on GHpages sites, which would correspond to ~10,000 users visiting the site and fetching all of the item sprites (~10 Mb). I think most of these would have to be unique users, too, since browsers should be caching the images. I don't expect we'll have to worry about that kind of volume.

An option for reducing bandwidth might be to have separate, lower-res versions of the item sprites to be used in the dropdowns.

**Comment by Ericzklm** [comment]
> We're a little pressed for space there at the moment, so it might take widening that area or moving some things around.

Something that can be tried is making the raid boss build 1/4 width like the normal raider builds and expanding the move select to 3/4 width. This can also give some extra vertical space.

I'll look into generating lower-res icon sprites that can be used in dropdown menus.

**Comment by Ericzklm** [comment]
I've tried removing the learn method from the dropdown and only having it show in pretty-mode since it's less important in edit-mode.
image
image


A possible way to preserve this information in edit-mode without as much clutter is to use "Helping Hand*" and "Growth**" without any extra icons.

**Comment by theastrogoth** [comment]
> I've tried removing the learn method from the dropdown and only having it show in pretty-mode since it's less important in edit-mode. 

I think this looks good! Having move types is likely the most helpful thing in edit-mode. I wonder if we can sneak in BP anywhere (probably a different PR for this).

> A possible way to preserve this information in edit-mode without as much clutter is to use "Helping Hand*" and "Growth**" without any extra icons.

Yet another option for edit-mode would be to use the type icons only if there isn't a relevant learn method icon to use. This would preserve type information for all moves except for egg moves. If we really want to go overboard, I suppose we could color code the egg spots, but I'm not sure it would look that good in practice.


> > We're a little pressed for space there at the moment, so it might take widening that area or moving some things around.
> 
> Something that can be tried is making the raid boss build 1/4 width like the normal raider builds and expanding the move select to 3/4 width. This can also give some extra vertical space.

I'm a little hesitant to use anything larger than a 1/2 width for the sake of having a sensible layout on mobile. We definitely have some room to shrink the Boss' build area, though.

**Comment by Ericzklm** [comment]
> I wonder if we can sneak in BP anywhere (probably a different PR for this)

I like the idea of having BP listed somewhere, maybe a mouse-over modal that can show things like BP, accuracy, physical/special/status, move effect, learn method, etc. A similar idea would be to add a mouse-over modal for the Pokemon list to see base stats, abilities, etc.

> Yet another option for edit-mode would be to use the type icons only if there isn't a relevant learn method icon to use.

It can be hard to tell what type a TM is since some of them are very close in color; to represent types, I think its best to have the type icons while representing the learn method some other way.

File: src/uicomponents/StatRadarPlot.tsx [hunk 1]
```diff
@@ -62,6 +66,19 @@ function StatRadarPlot({nature, evs, stats, bossMultiplier=100}: {nature: Nature
...
+                        line: {
+                            width: 1.5,
+                            color: isDark ? "#cccccc" : "#888888",
+                        },
+                        name: "axis",
+                        r: [1, 1, 1, 1, 1, 1, 1],
+                        theta: ticktexts,
+                        fill: "toself",
+                        fillcolor: "rgba(0, 0, 0, .1)",
+                    },
```
Comment (theastrogoth) [inline src/uicomponents/StatRadarPlot.tsx]: A very minor thing, but it looks like the corners of the hexagon get very slightly cut off (presumably by half of their line width). If we want sharp corners, we could change `r` here to be a little smaller than 1, or we could edit the limits of the plot to be a little greater than 1.

**Comment by theastrogoth** [comment]
This looks good to me! The additional icons are quite helpful, even more than I expected. 

> I like the idea of having BP listed somewhere, maybe a mouse-over modal that can show things like BP, accuracy, physical/special/status, move effect, learn method, etc. A similar idea would be to add a mouse-over modal for the Pokemon list to see base stats, abilities, etc.

A mouse-over modal sounds great! I think it would be pretty easy to throw a [MUI Popper](https://mui.com/material-ui/react-popper/) or something on top of what we have.

> It can be hard to tell what type a TM is since some of them are very close in color; to represent types, I think its best to have the type icons while representing the learn method some other way.

That makes sense. I agree that the type icons are the best option for the dropdown. Perhaps the learn method could have a spot in the modal, as well as keep their spot in the pretty-mode display.

File: src/uicomponents/StatRadarPlot.tsx (hunk 1 above)
Comment (theastrogoth) [reply src/uicomponents/StatRadarPlot.tsx]: Looks good!

**Comment by theastrogoth** [comment]
I love the placeholder, by the way

**Pull request merged**